convert_swig doesn't generate %template directives for instances of template types that use a default member initializer

The `template_var_def` regular expression in `convert_swig` does not account for [default member initializer](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization) syntax and thus fails to generate the necessary `TRICK_SWIG_TEMPLATE` lines to instantiate the template to SWIG via `%template`. Without those lines, SWIG knows nothing about a template's assignment operator. If the template's `operator=` is `private`, the generated `*_py.cpp` for any class containing an instance of that template and using default member initialization will fail to compile.

Foo.hh
```c++
template <class T>
class Foo {

    public:
    Foo() {}

    private:
    void operator=(const Foo&);

};
```

S_define
```c++
#include "sim_objects/default_trick_sys.sm"
##include "Foo.hh"

class Sandbox : public Trick::SimObject {

    public:
    Foo<int>foo{}; // default member initializer

    private:
    void operator=(const Sandbox&);

};

Sandbox sandbox;
```

S_source_py.i
```c++
class Sandbox : public Trick::SimObject {

    public:
    Foo<int>foo{};
********** There should be a TRICK_SWIG_TEMPLATE line here **********

    private:
    void operator=(const Sandbox&);



    public:
        virtual int call_function( Trick::JobData * curr_job ) ;
        virtual double call_function_double( Trick::JobData * curr_job ) ;
};
#define TRICK_SWIG_DEFINED_Sandbox
```

trick-CP
```shell
In file included from /users/dbankier/trick/sims/SIM_sandbox/S_source.hh:27:0,
                 from build/users/dbankier/trick/sims/SIM_sandbox/S_source_py.cpp:3623:
/users/dbankier/trick/sims/SIM_sandbox/models/Foo.hh: In function ‘PyObject* _wrap_Sandbox_foo_set(PyObject*, PyObject*)’:
/users/dbankier/trick/sims/SIM_sandbox/models/Foo.hh:6:10: error: ‘void Foo<T>::operator=(const Foo<T>&) [with T = int]’ is private
     void operator=(const Foo&);
          ^
build/users/dbankier/trick/sims/SIM_sandbox/S_source_py.cpp:16058:25: error: within this context
   if (arg1) (arg1)->foo = *arg2;
```

--- a/libexec/trick/convert_swig
+++ b/libexec/trick/convert_swig
@@ -94,7 +94,7 @@ my $template_def   = qr/template\s*                        # keyword template
                /sx ;
 my $template_var_def  = qr/(?:\:\:)?[_A-Za-z][:\w]*\s*     # template name
             <[\w\s\*,:<>]*>\s*                             # template parameters
-            [_A-Za-z]\w*\s*;                               # var name ;
+            [_A-Za-z]\w*\s*(?:[{=].*?)?;                   # var name ;
                /sx ;
 
 # This list is the list of all STL types to ignore.
@@ -619,11 +619,9 @@ sub process_class($$$$$) {
         if ( $template_var_def_str ne "" )  {
             # if there is a whiff of const in the template we are punting.
             if ( $template_var_def_str !~ /^const|[<,\s]const\s/ ) {
-                #print "*** template_var = $template_var_def_str ***\n" ;
-                $template_var_def_str =~ /(.*?)([_A-Za-z]\w*)\s*;/s ;
+                $template_var_def_str =~ /(.*?>)\s*([_A-Za-z]\w*).*?;/s ;
                 my ($template_full_type) = $1 ;
                 my ($var_name) = $2 ;
-                #print "*** var_name = $var_name ***\n" ;
                 $$new_contents_ref .= $template_var_def_str ;
 
                 $template_full_type =~ /([_A-Za-z][:\w]*)\s*</ ;

--- a/test/SIM_default_member_initializer/S_define
+++ b/test/SIM_default_member_initializer/S_define
@@ -1,0 +1,20 @@
+#include "sim_objects/default_trick_sys.sm"
+##include "Foo.hh"
+
+class Sandbox : public Trick::SimObject {
+
+    public:
+    // These should get TRICK_SWIG_TEMPLATE lines
+    Foo<int> foo{1};
+    Foo<int> foo2 = {2};
+    Foo<int> foo3;
+
+    // Functions should not
+    template<class T> void bar();
+
+    private:
+    void operator=(const Sandbox&);
+
+};
+
+Sandbox sandbox;

--- a/test/SIM_default_member_initializer/S_overrides.mk
+++ b/test/SIM_default_member_initializer/S_overrides.mk
@@ -1,0 +1,1 @@
+TRICK_CXXFLAGS += -Imodels -std=c++11

--- a/test/SIM_default_member_initializer/models/Foo.hh
+++ b/test/SIM_default_member_initializer/models/Foo.hh
@@ -1,0 +1,15 @@
+template <class T>
+class Foo {
+
+    public:
+
+    Foo() {}
+
+    Foo(int i) {
+        (void)i;
+    }
+
+    private:
+    void operator=(const Foo&);
+
+};

--- a/test/makefile
+++ b/test/makefile
@@ -1,8 +1,22 @@
 TRICK_HOME := $(abspath $(dir $(lastword $(MAKEFILE_LIST)))..)
 export TRICK_HOST_CPU := $(shell $(TRICK_HOME)/bin/trick-gte TRICK_HOST_CPU)
 
-# List out sims we want to compile
-COMPILE_DIRS = \
+SIMS_TO_COMPILE_ONLY = \
+	SIM_alloc_test \
+	SIM_default_member_initializer \
+	SIM_demo_inputfile \
+	SIM_exclusion_mechanisms \
+	SIM_isystem \
+	SIM_measurement_units \
+	SIM_parse_s_define \
+	SIM_target_specific_variables \
+	SIM_test_abstract \
+	SIM_test_inherit \
+	SIM_test_ip2 \
+	SIM_threads_simple \
+	SIM_trickcomm
+
+SIMS_TO_COMPILE_AND_RUN = \
 	SIM_demo_sdefine \
 	SIM_events \
 	SIM_python_namespace \
@@ -16,20 +30,6 @@ COMPILE_DIRS = \
 	SIM_test_templates \
 	SIM_threads
 
-SIMS_NEEDING_TEST = \
-	SIM_alloc_test \
-	SIM_demo_inputfile \
-	SIM_exclusion_mechanisms \
-	SIM_isystem \
-	SIM_measurement_units \
-	SIM_parse_s_define \
-	SIM_target_specific_variables \
-	SIM_test_abstract \
-	SIM_test_inherit \
-	SIM_test_ip2 \
-	SIM_threads_simple \
-	SIM_trickcomm
-
 # Sims with problems, no purpose, or maybe shouldn't be a test
 # SIM_leaks ( should be deleted )
 # SIM_test_varserv  ( not sure what it is testing )
@@ -39,20 +39,20 @@ SIMS_NEEDING_TEST = \
 # This test is temporarily sitting out until fixed.
 #    SIM_test_varserv
 
-EXECUTABLES = $(addsuffix /T_main_${TRICK_HOST_CPU}_test.exe, $(COMPILE_DIRS) $(SIMS_NEEDING_TEST))
-UNIT_TEST_RESULTS = $(addprefix $(TRICK_HOME)/trick_test/, $(addsuffix .xml, $(COMPILE_DIRS)))
+EXECUTABLES = $(addsuffix /T_main_${TRICK_HOST_CPU}_test.exe, $(SIMS_TO_COMPILE_AND_RUN) $(SIMS_TO_COMPILE_ONLY))
+UNIT_TEST_RESULTS = $(addprefix $(TRICK_HOME)/trick_test/, $(addsuffix .xml, $(SIMS_TO_COMPILE_AND_RUN)))
 
 test: $(EXECUTABLES) $(UNIT_TEST_RESULTS) data_record_results
 
 clean:
 	rm -f $(UNIT_TEST_RESULTS)
-	- for i in $(COMPILE_DIRS) ; do \
+	- for i in $(SIMS_TO_COMPILE_AND_RUN) ; do \
 		if [ -f "$$i/"[Mm]"akefile" ] ; then \
 			$(MAKE) -C $$i spotless ; \
 		fi \
 	done
 
-	- for i in $(SIMS_NEEDING_TEST) ; do \
+	- for i in $(SIMS_TO_COMPILE_ONLY) ; do \
 		if [ -f "$$i/"[Mm]"akefile" ] ; then \
 			$(MAKE) -C $$i spotless ; \
 		fi \

